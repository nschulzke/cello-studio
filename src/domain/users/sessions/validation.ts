import { fromBool } from 'domain/helpers/Result';

var emailRegex = /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;

export const emailIsValid = ({ email }) =>
  fromBool(emailRegex.test(String(email)), 'Email is invalid.');
