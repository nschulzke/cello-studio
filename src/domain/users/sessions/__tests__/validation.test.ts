import * as validation from '../validation';

describe('validation', () => {
  describe('emailIsValid', () => {
    it('accepts a valid email address', () => {
      expect(validation.emailIsValid({ email: 'test@test.com' }).success).toBe(true);
    });

    it('rejects an invalid email address', () => {
      let failure = validation.emailIsValid({ email: 'testtest.com' });
      expect(failure.success).toBe(false);
      expect(failure.data).toMatchSnapshot();
    });
  });
});
