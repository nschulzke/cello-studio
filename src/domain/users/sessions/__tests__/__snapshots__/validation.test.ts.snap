// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`validation emailIsValid rejects an invalid email address 1`] = `"Email is invalid."`;
